### DISCUSSION TITLE
Laps

### DISCUSSION THREAD

**Discussion by kenlpeters** [philosowaffle/peloton-to-garmin#176] [Q&A]
When the data gets imported into Garmin the laps are weird. It seems they are structured as warm up, run, cool down or some other metric with rides. Given most folks in US consider a lap 1 mi can the laps be imported as 1 mi?

**Comment by philosowaffle** [comment]
There are 2 use cases that P2G currently handles.

1. Peloton breaks each class into [Segments](https://github.com/philosowaffle/peloton-to-garmin/blob/master/UnitTests/Data/cycling_workout.json#L241) and reports the rollup metrics for those segments. By default these are the laps that are created and in most cases will correspond with Warm Up, Ride, Cool Down portions of the class.  To me, this made more sense than using Distance as a lap designator since Peloton classes can have different exercise types within them, such as a Weights section in the middle of a ride.

2. If the ride reports [Target Cadence information](https://github.com/philosowaffle/peloton-to-garmin/blob/master/UnitTests/Data/cycling_workout.json#L11212), then the laps will correspond with each change in Cadence Target. From a lap perspective, this is not really ideal, however this is how the [FIT spec](https://developer.garmin.com/fit/cookbook/encoding-workout-files/) expects the Target metrics to be documented and it is the only way to get the Cadence chart to reflect the targets as seen [here](https://github.com/philosowaffle/peloton-to-garmin/blob/master/images/cadence_target.png).

If we were to implement creating distance based laps it would definitely have to support both miles and kilometers as we have [P2G users](https://github.com/philosowaffle/peloton-to-garmin/issues/15) on both measurement systems.

**Reply by philosowaffle** [reply]
And we would probably need a config option to designate preference since only one of these 3 strategies could be used at a time. i.e.

1. Class Segment based Laps
1. Cadence Target based Laps
1. Distance based Laps

**Reply by kenlpeters** [reply]
I think for riding your logic is sound but for running it is all about pace. Maybe have flag only for running to show pace by lap/mile. If that is possible.

**Reply by philosowaffle** [reply]
[Feature request](https://github.com/philosowaffle/peloton-to-garmin/issues/177) created, continue any further discussion or feedback on the issue itself.
